webhook/d7_module/quicksilver/quicksilver.module

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * @file
+ * Provides a hook for Pantheon workflow.
+ */
+
+/**
+ * Implements hook_permission().
+ */
+function quicksilver_permission() {
+  return array(
+    'administer quicksilver' => array(
+      'title' => t('Administer Quicksilver'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_menu
+ */
+function quicksilver_menu() {
+  $items = array();
+  $items['admin/config/services/quicksilver'] = array(
+    'title' => 'Pantheon Quicksilver',
+    'description' => 'Configure API key and test mode for the Pantheon quicksilver module.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('quicksilver_admin_form'),
+    'access arguments' => array('administer quicksilver'),
+    'file' => 'quicksilver.admin.inc',
+  );
+  $items['quicksilver/%'] = array(
+    'page callback' => 'quicksilver_webhook_handler',
+    'access callback' => TRUE,
+    'delivery callback' => 'drupal_json_output'
+  );
+  return $items;
+}
+
+/**
+ * quicksilver webhook callback.
+ */
+function quicksilver_webhook_handler() {
+  $success = FALSE;
+  $message = '';
+  $submitted_key = arg(1);
+  $api_key = variable_get('quicksilver_api_key', '');
+
+  if (isset($submitted_key) && $submitted_key == $api_key) {
+    $success = TRUE;
+    $data = array(
```

Comment:
Maybe $data should just be $_POST.  The module will be able to read $_ENV directly.  I don't think that `$_POST['payload']` is correct.